tests/timeseries/mock_microgrid.py

```diff
@@ -71,28 +73,76 @@ def __init__(  # pylint: disable=too-many-arguments
                 to.  Useful in tests where multiple namespaces (logical_meter,
                 battery_pool, etc) are used, and the same metric is used by formulas in
                 different namespaces.
+            graph: optional, a graph of components to use instead of the default grid
+                layout. If specified, grid_meter must be None.
         """
-        self._components: Set[Component] = set(
-            [
-                Component(1, ComponentCategory.GRID),
-            ]
+        if (grid_meter is not None or grid_meter is True) and graph is not None:
```

Comment:
In `(grid_meter is not None or grid_meter is True)`, the second condition appears to be redundant.  
```suggestion
        if grid_meter is not None and graph is not None:
```